changedetectionio/model/Watch.py

```diff
@@ -178,3 +182,23 @@ def lines_contain_something_unique_compared_to_history(self, lines=[]):
                 return True
 
         return False
+
+    # Get diff types (addition, deletion, modification) from the previous snapshot and new_text
+    # uses similar algorithm to customSequenceMatcher in diff.py
+    # Returns a dict of diff types and wether they are present in the diff
+    def get_diff_types(self, new_text):
+        
+        diff_types = {
+            'add': False,
+            'del': False,
+        }
+
+        # get diff types using difflib
+        cruncher = difflib.SequenceMatcher(isjunk=lambda x: x in " \\t", a=self.get("previous_text", ""), b=new_text)
+        for tag, alo, ahi, blo, bhi in cruncher.get_opcodes():
```

Comment:
im getting a exception when it hits this line
```
object of type 'bool' has no len()
```

Reply:
Yeah I found the problem. The default value in the model was set to False. I changed to a empty string. Changes coming now